### PR TITLE
Add Documentation for Pragmas Plugin

### PR DESCRIPTION
**Pull request by fendor**

### CODE REVIEW

File: src/Haskell/Ide/Engine/Plugin/Pragmas.hs [hunk 1]
```diff
@@ -53,15 +58,22 @@ addPragmaCmd = CmdSync $ \(AddPragmaParams uri pragmaName) -> do
 
 -- ---------------------------------------------------------------------
 
+-- | Offer to add a missing Language Pragma to the top of a file.
+-- Pragmas are defined by a curated list of known pragmas, see 'possiblePragmas'.
+-- May also offer to add unknown pragmas, if the pragma is a subexpression of
+-- an existing plugin.
```
Comment (alanz) [inline src/Haskell/Ide/Engine/Plugin/Pragmas.hs]: I do not understand the last two lines, can you try use a different wording.

File: src/Haskell/Ide/Engine/Plugin/Pragmas.hs [hunk 2]
```diff
@@ -71,13 +83,16 @@ codeActionProvider plId docId _ (J.CodeActionContext (J.List diags) _monly) = do
...
 
+-- | Find all Pragmas that have the search term as infix.
 findPragma :: T.Text -> [T.Text]
 findPragma str = concatMap check possiblePragmas
   where
     check p = [p | T.isInfixOf p str]
 
 -- ---------------------------------------------------------------------
 
+-- | Possible Pragama names.
```
Comment (alanz) [inline src/Haskell/Ide/Engine/Plugin/Pragmas.hs]: Typo, `pragama`

File: src/Haskell/Ide/Engine/Plugin/Pragmas.hs (hunk 1 above)
Comment (fendor) [reply src/Haskell/Ide/Engine/Plugin/Pragmas.hs]: It was wrong anyways, thanks for making me look at it again.

File: src/Haskell/Ide/Engine/Plugin/Pragmas.hs [hunk 3]
```diff
@@ -71,13 +81,16 @@ codeActionProvider plId docId _ (J.CodeActionContext (J.List diags) _monly) = do
 
 -- ---------------------------------------------------------------------
 
+-- | Find all Pragmas are an infix of the search term.
 findPragma :: T.Text -> [T.Text]
 findPragma str = concatMap check possiblePragmas
```
Comment (fendor) [inline src/Haskell/Ide/Engine/Plugin/Pragmas.hs]: this could be just
`findPragma str = filter (\p -> T.isInfixOf p str) possiblePragmas` right?

Comment (alanz) [reply src/Haskell/Ide/Engine/Plugin/Pragmas.hs]: I think so.  But that change should be in a separate PR, this one is documentation only

**Pull request merged**